tests/test_polaraveraging.py

```diff
@@ -0,0 +1,225 @@
+                            rtol, err_msg="Failure when lens and a source share an RA")
+    testing.assert_allclose(pa._compute_theta_phi(ra_l, dec_l, ra_s, np.array([51.49, 51.55])),
+                            [[0.00032601941539388962, 0.00106951489719733675], [0.00000000000000000000, 1.77544123918164542530]],
+                            rtol, err_msg="Failure when lens and a source share a DEC")
+    # l/s at same ra AND dec
+    testing.assert_raises(ValueError, pa._compute_theta_phi, ra_l, dec_l, np.array([161.32, 161.34]), np.array([51.49, 51.55]))
+
+    #testing.assert_warns(UserWarning,pa._compute_theta_phi, ra_l, dec_l, np.array([151.32, 161.34]), np.array([41.49, 51.55]),'flat')
+    
+    #testing.assert_allclose(pa._compute_theta_phi(ra_l, dec_l, np.array([161.32, 161.34]), np.array([51.49, 51.55])),
+    #                        [[0.00000000000000000000, 0.00106951489719733675], [0.00000000000000000000, 1.77544123918164542530]],
+    #                        rtol, err_msg="Failure when lens and a source share an RA and a DEC")
+    
+    
+    testing.assert_allclose(pa._compute_theta_phi(0.1, dec_l, np.array([359.9,180.1]), dec_s),
+                            [[2.37312589, 1.95611677], [-2.94182333e-04,  3.14105731e+00]],
+                            rtol, err_msg="Failure when ra_l and ra_s are close but on the two sides of the 0 axis")   
+
+    testing.assert_allclose(pa._compute_theta_phi(0, dec_l, np.array([359.9,180.1]), dec_s),
+                            [[2.37203916, 1.9572035], [-2.94317111e-04,  3.14105761e+00]],
+                            rtol, err_msg="Failure when ra_l and ra_s are separated by pi + epsilon")  
+    
+    testing.assert_allclose(pa._compute_theta_phi(-180, dec_l, np.array([180.1,-90]), dec_s),
+                            [[2.36986569, 0.97805881], [-2.94587036e-04,  3.14052196e+00]],
+                            rtol, err_msg="Failure when ra_l and ra_s are the same but one is defined negative")     
+    
+    testing.assert_allclose(pa._compute_theta_phi(ra_l, 90, ra_s, np.array([51.45,-90])),
+                            [[0.67282443, 3.14159265], [-1.57079633, -1.57079633]],
+                            rtol, err_msg="Failure when dec_l and dec_s are separated by 180 deg")  
+
+
+def test_compute_shear():
+    # the numbers on this test must be revise, output values were extracted from the first run of the code
+    g1, g2 = np.ones(2), np.ones(2)
+    rtol = 1e-6
+    ra_l, dec_l = 161.32, 51.49
+    ra_s, dec_s = np.array([161.29, 161.34]), np.array([51.45, 51.55])
+    testing.assert_allclose(
+        pa._compute_shear(ra_l, dec_l, ra_s, dec_s, g1, g2, sky='flat'),
+        np.array([[0.00077050407583119666, 0.00106951489719733675],
+            [ 1.408693e+00,  1.315362e+00],
+            [ 1.248319e-01, -5.194458e-01]]
+            ),
+        rtol=rtol)
+
+def test_gc_wrapper():
```

Comment:
These are testing the monkey-patched versions of the functions. We can just axe this functions and move these tests to their appropriate functions. i.e. We should be testing that `compute_shear(Cluster, stuff)` works and `Clusters().compute_shear()` works together in the tests for `compute_shear()`.

In the above example, we can test `compute_shear(Cluster, stuff)` extensively and then just test `compute_shear(Cluster, stuff) == Cluster().compute_shear(stuff)`